### DISCUSSION TITLE
Way to implement empty_as_undefined ?

### DISCUSSION THREAD

**Discussion by CaptainDriftwood** [wyfo/apischema#389] [Q&A]
Is there a way to have empty collections be treated as Undefined upon serialization? In the same vein, is there a way to do this with empty/blank strings? (I know that str also implement the __len__ method, curious if a default serialization could treat strings differently from collections in this regard)

**Comment by wyfo** [comment]
I think you can achieve that in two different ways:
- [conditional field skipping](https://wyfo.github.io/apischema/0.17/data_model/#skip-field-serialization-depending-on-condition)
- [serialized method/property](https://wyfo.github.io/apischema/0.17/de_serialization/#serialized-methodsproperties) (returning Undefined) with the same alias than the field (which will then be overridden).

**Reply by CaptainDriftwood** [reply]
Both those options have to be used at the field level on a dataclass. Is there no way to declare this type of behavior at the class level, or globally via a setting?
